hw/ip/clkmgr/data/clkmgr.prj.hjson

```diff
@@ -10,7 +10,7 @@
     sw_checklist:       "/sw/device/lib/dif/dif_clkmgr",
     revisions: [
       {
-        version:            "0.1",
+        version:            "1.0",
```

Comment:
correct me if i'm wrong, i thought when we made a version change we were also required to capture a history of the last version's commit id?

Reply:
According to https://docs.opentitan.org/doc/project/development_stages/#versioning that's correct:

> The commit ID of the version that was signed off is recorded in the commit_id field of the .prj.hjson file.

Reply:
I think this is only supposed to be needed if we're dropping L/D/V/S levels. The idea is that you want someone to be able to find "the old version that worked". So I *think* this is all good.

(Note: I've just noticed how close our terminology is to being "[LVDS](https://en.wikipedia.org/wiki/Low-voltage_differential_signaling)". If only we did verification before design... :-P)

Reply:
If I read the docs correctly, there are basically 2 transitions where the commit ID **must** be recorded:
1. signoff and move to L2 (see [Versioning](https://docs.opentitan.org/doc/project/development_stages/#versioning) as pointed out by @a-will)
2. regression in the D/V stages (see [multiple-versions-of-a-design](https://docs.opentitan.org/doc/project/development_stages/#multiple-versions-of-a-design) as pointed out by @rswarbrick)

For all other transitions, recording the commit ID appears to be [optional](https://docs.opentitan.org/doc/project/development_stages/#commit-id).

As for keeping the revision history by creating another entry in the hjson like in this example here:
https://github.com/lowRISC/opentitan/blob/e9c6fcce94aa3fae40ecd5388710abd1720c7fc5/hw/ip/gpio/data/gpio.prj.hjson#L12-L26

I believe this is only required if we [regress](https://docs.opentitan.org/doc/project/development_stages/#multiple-versions-of-a-design) (i.e., move backwards in terms of D/V or life stages).

However, we haven't been super consistent about recording commits / creating another hjson entry in the past when we bumped the version number. 

Looking at the [hw dashboard](https://docs.opentitan.org/hw/), several transitions where this has been done were downgrades from L2 -> L1:
- gpio
- hmac
- rv_timer
- uart
- tlul

But there appear to be others as well:
- flash_ctrl
- otbn
- otp_ctrl
- pwrmgr
- spi_device

It is hard to tell whether these were due to version number increases only, or due to regressions in D/V stages that by now may have progressed past the stages before the downgrade.

I am happy to add the commits / version snapshots if we think this is useful - but if I interpret the process right, I think this would be optional in this case...

Maybe this is something that needs some more clarification in the docs?